lib/index.d.ts

```diff
@@ -0,0 +1,801 @@
+import type { Response } from "node-fetch"
+export { Response }
+
+export interface DinkyRequestOptions {
+  /**
+   * User's personal API key taken from their account.
+   */
+  key?: string
+
+  /**
+   * An ID of a filter found on filters page.
+   * The ID can be found on filters page: <https://derpibooru.org/filters>
+   */
+  filter?: number
+}
+
+export class Dinky {
+  /**
+   * Creates an instance of derpibooru API client.
+   * See `DinkyRequestOptions` for more details on configuration.
+   *
+   * Default options: `{}`
+   */
+  constructor(options?: DinkyRequestOptions)
+
+  /**
+   * Creates a request handler for `/api/v1/json/search`
+   *
+   * @param query – query params
+   */
+  search(...query: (string | string[])[]): ImagesSearch
+
+  /**
+   * Creates a request handler for `/api/v1/json/comments`
+   */
+  comments(): Comments
+
+  /**
+   * Creates a request handler for `/api/v1/json/tags`
+   */
+  tags(): Tags
+
+  /**
+   * Creates a request handler for `/api/v1/json/images`
+   */
+  images(): Images
+
+  /**
+   * Creates a `Lists` class that contains a bunch of shortcuts for `Search`
+   */
+  lists(): Lists
+}
+
+export function dinky(options: DinkyRequestOptions): Dinky
+export default dinky
+
+export class NetworkError extends Error {
+  response: Response
+  url: string
+  status: number
+  statusText: string
+
+  // TODO: should this one be private?
+  constructor(message: string, response: Response)
+}
+
+interface Request<T> {
+  /**
+   * Sets the page offset.
+   *
+   * Default offset: `1`
+   */
+  page(offset: number): this
+
+  /**
+   * Executes current request.
+   *
+   * Default options: `{}`
+   */
+  exec<T>(options?: DinkyRequestOptions): Promise<T>
+
+  then: Promise<T>["then"]
+  catch: Promise<T>["catch"]
+}
+
+// TODO: its constructor is marked as public, so why does it have @private annotation then?
+// I guess this doesn't mean it is public to external users, but public inside of this lib or smth...
+interface Entities<Entity, EntitiesPage> extends Request<EntitiesPage> {
+  /**
+   * Creates a new `Search` request for given query
+   */
+  search(query: (string | string[])[]): Search<EntitiesPage>
+
+  /**
+  * Get one entity by its ID
+  */
+  getById(id: number, options?: DinkyRequestOptions): Promise<Entity>
+}
+
+// TODO: Constructors of these types are not public, but they are described
+// in README, this is weird
```

Comment:
They are mentioned with description only. This is due to documentation format I've used for my projects : D

Reply:
I dunno what to say :D...